grpc/client.go

```diff
@@ -66,9 +78,13 @@ func (w *grpcClientWrapper) Start(ctx context.Context, opts ...pubsub.Option) (<
 		}
 		for {
```

Comment:
It is not required to duplicate this logic. From my point of view it would be better to implement gracefully shutdown from ddi side, and just use cancelation of the context to gracefully exit from pubsub loop

Reply:
The signal handling will be needed for every thread that functions in a loop.  As such, we will also be doing the code on our loops but you need to do it on the pubsub loops as well since they are separate threads.

Reply:
I believe what @burov is saying is correct. A client should only manage its lifetime through a context by properly handling `Done()` messages. Signal handling as well as notifying a client about the signals is caller's responsibility (e.g. by calling `ctx.Done()`).

It's up to a caller to use the proper context (deadline/timeout/both). The caller is also responsible for closing/cancelling the open contexts (e.g. upon receiving a signal). The client is only responsible for handling those cancellations gracefully.

With the respect to the above, I'm not sure if doing just `return` for `ctx.Done()` is graceful enough. At the very minimal the connections/streams should be gracefully closed.

In the pub-sub server code however those signals should be handled as pointed by @rvinfoblox.

Reply:
Agreed. The client could handle the signals and then call `Done()`.  

@burov, Is pub-sub server code currently handling signals?

Reply:
@rvinfoblox yes, I added signals handling for service.